src/common/colorspaces.c

```diff
@@ -161,7 +160,11 @@ static int dt_colorspaces_get_matrix_from_profile(cmsHPROFILE prof, float *matri
   // so the only right way to handle it is to let LCMS apply it.
   const int UsedDirection = input ? LCMS_USED_AS_INPUT : LCMS_USED_AS_OUTPUT;
 
-  if(cmsIsCLUT(prof, intent, UsedDirection)) return 1;
+  if(cmsIsCLUT(prof, INTENT_PERCEPTUAL, UsedDirection)
```

Comment:
If I read correctly this patch the only change is there. Instead of checking for a specific intent we check for any intent if there is a CLUT or not.

The consequence is that we don't need to pass intent here and in cascade we have many places where intent parameter is removed.

Right?

If so please add some comment here saying exactly that. We return early because there is a CLUT for an intent as described in your commit log. This is delicate area, not every people are expert on this (count myself here) and so a comment will avoid any confusion when reading the code.

TIA.

Reply:
I amended the previously existing comment that briefly described the matter. Now there's a better explanation of what a swapped matrix means in this case. Would this be sufficient?

Swapped matrix was a new concept for me also when I first read this part of the code. Most of the references to swapped matrix seem to point to DisplayCAL - in fact, swapped matrices are mentioned on the front page: https://displaycal.net/ (search for "swapped matrix")